test/test_torch.py

```diff
@@ -19438,23 +19442,23 @@ def inner(self, device, dtype):
     ('pow', 'tensor', _small_3d, lambda t, d: [_small_3d(t, d).abs()],
         1e-1, 1e-1, 1e-5, _float_types2),
     ('addbmm', '', _small_2d, lambda t, d: [_small_3d(t, d), _small_3d(t, d)],
-        1e-1, 1e-1, 1e-4, _float_types2, _cpu_types, True, [tf32_on_and_off(0.005)]),
+        1e-1, 1e-1, 1e-4, _complex_and_float_types2, _cpu_types + _complex_types, True, [tf32_on_and_off(0.005)]),
```

Comment:
@mruberry IIUC, `_cpu_types` is empty list, and `_cpu_types + _complex_types` would mean run CPU test only with complex?

Reply:
Boy does this set of tests need to be entirely rewritten. 

Your understanding is correct. Note that "testing" on CPU here just means verifying that the function and method variants of the op produce the same value. There's no check those values are correct.

Reply:
I didn't know that the cpu tests here only compare the function and method result which basically doesn't test anything for a lot of ops. will remove it.

Reply:
"testing" on cpu means results in low precision (bfloat16 or whatever _cpu_types is) is compared to result in fp32 precision. For _cpu_types=fp32 it would indeed just compare function and method.